### ISSUE TITLE
ebs_optimization defaults.

### ISSUE THREAD

**Issue by bkochendorfer** [terraform-providers/terraform-provider-aws#2667]
### Terraform Version
```bash
% terraform -v
Terraform v0.11.1
+ provider.aws v1.5.0
+ provider.template v1.0.0
```

### Affected Resource(s)
- aws_launch_configuration
- aws_instance

This issue affects multiple resources. I listed those I am aware of but I believe it will affect any resource which passes the `ebs_optimized` boolean.

### Expected Behavior
I would expect that the `ebs_optimized` option when not specified would default to AWS settings rather than defaulting to false. For example on the [AWS EBS User Guide](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html) the `m4.large` instance type will default to using ebs optimization. 

### Actual Behavior
Terraform appears to default to false for this boolean rather than the defaults provided by AWS.

### Steps to Reproduce
Fairly simple, create an instance without `ebs_optimization` specified and verify if it using the defaults provided by AWS.

### References
-[AWS EBS User Guide](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html)

**Comment by ericofusco** [comment]
@bkochendorfer,

EBS-optimized setting is tricky, this is not an issue with terraform.

EBS-optimized is set with false if EbsOptimized is not defined (Reference: http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html), the same behavior occur with **all** instances, including the ones that are EBS-optimized by default.

Instances that are EBS-optimized by default will work with EBS-optimized **even when EBS-optimized is set to false.**

> The current generation instance types are EBS–optimized by default. For those instances, there is no need to enable EBS optimization and no effect if you disable EBS optimization.

Reference: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html

So basically the setting ebs_optimized (EbsOptimized on AWS API) only applies for instances that are not EBS-Optimized by default.

I understand this can be confusing, but this is an issue with AWS, I think they should show EbsOptimized with true for instances that are EBS-Optimized by default, regardless of the parameter given if it doesn't have any effect accordingly to the documentation.

**Comment by apparentlymart** [comment]
Hi @bkochendorfer! Thanks for reporting this.

Thanks also for that additional context, @bkochendorfer. Indeed it seems like this is an API inconsistency that is therefore surfacing as a  Terraform inconsistency.

In an ideal world Terraform would be able to recognize this situation and compensate, but unfortunately I think in this case we cannot reasonably do that since the set of available instance types changes frequently and we would not want to require a new AWS release to access newly-added instance types with the correct behavior.

Given that, I think the current behavior is the best we can do even though it's not ideal. However, I think we could at least update the documentation for the resources in question to duplicate that caveat from the EC2 documentation in Terraform's terms, so it's more obvious to those who primarily reference the Terraform documentation and may not be familiar with the details of the underlying service.

**Comment by merlineus** [comment]
Hi @apparentlymart ,

Current behavior leads to instance recreation if you change instance type via AWS console as EBS optimized parameter became "true" (for example from c5.large to c5.xlarge).

So I think better way to default this setting to "true" in AWS provider for instances that is EBS optimized by default.

**Comment by deepak-goel** [comment]
As a workaround, you can use `ignore_changes` inside the `aws_instance` resource block. Something like

```        data =  {
            "aws_instance": {
                self.name() : {
                    "ami"   : IMAGE_ID[self.env],
                    "instance_type" : self.instance_flavor(),
                    "count" : self.min_instance_count,
                    "associate_public_ip_address" : "false",
                    "availability_zone": "${data.aws_availability_zones.available.names[(count.index)%2]}",
                    "subnet_id": "${var.aws_subnet[(count.index)%2]}",
                    "tags" : {
                        'Name' : instance_name(),
                        'component' : self.component,
                        'subcomponent' : self.subcomponent,
                        'env' : self.env
                    },
                    **"lifecycle" : {
                        "ignore_changes" : [ "ebs_optimized" ]
                    }**,
                    "key_name" : "test",
                    "vpc_security_group_ids" : [ "sg-xxx","sg-xxx","sg-xxx"]
                }
            }
        }```
